mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-google/google/fwmodels"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+	"github.com/hashicorp/terraform-provider-google/google/fwutils"
+	"github.com/hashicorp/terraform-provider-google/google/fwvalidators"
+	"google.golang.org/api/iamcredentials/v1"
+)
+
+var _ ephemeral.EphemeralResource = &googleEphemeralServiceAccountIdToken{}
+
+func GoogleEphemeralServiceAccountIdToken() ephemeral.EphemeralResource {
+	return &googleEphemeralServiceAccountIdToken{}
+}
+
+type googleEphemeralServiceAccountIdToken struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account_id_token"
+}
+
+type ephemeralServiceAccountIdTokenModel struct {
+	TargetAudience       types.String `tfsdk:"target_audience"`
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	IncludeEmail         types.Bool   `tfsdk:"include_email"`
+	IdToken              types.String `tfsdk:"id_token"`
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema.Description = "This ephemeral resource provides a Google OpenID Connect (oidc) id_token."
+	resp.Schema.MarkdownDescription = "This ephemeral resource provides a Google OpenID Connect (oidc) id_token."
+
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"target_audience": schema.StringAttribute{
+				Description: "The audience claim for the `id_token`.",
+				Required:    true,
+			},
+			"target_service_account": schema.StringAttribute{
+				Description: "The email of the service account being impersonated.  Used only when using impersonation mode.",
+				Optional:    true,
+				Validators: []validator.String{
+					fwvalidators.ServiceAccountEmailValidator{},
+				},
+			},
+			"delegates": schema.SetAttribute{
+				Description: "Delegate chain of approvals needed to perform full impersonation. Specify the fully qualified service account name.  Used only when using impersonation mode.",
+				Optional:    true,
+				ElementType: types.StringType,
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(fwvalidators.ServiceAccountEmailValidator{}),
+				},
+			},
+			"include_email": schema.BoolAttribute{
+				Description: "Include the verified email in the claim. Used only when using impersonation mode.",
+				Optional:    true,
+				Computed:    true,
+			},
```

Comment:
This field is Optional with a Default of false in the data source we're mimicking: https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/resourcemanager/data_source_google_service_account_id_token.go

Please remove Computed, and ensure the logic of the Open method assumes false as the default value for this field

Reply:
When ValueBool is Null or Unknown it is [automatically set to false](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.13.0/types/basetypes#BoolValue.ValueBool)

So the default value is already covered here, I'll remove compute and include a note next to Optional explaining this.